### DISCUSSION TITLE
GitVersion Conversion

### DISCUSSION THREAD

**Discussion by AshleighAdams** [adamralph/minver#486] [Q&A] [answered]
Hey all,

First, I want to say thanks for this, I'd previously configured GitVersion to essentially do the same thing... more or less, but this works much better, especially when doing prereleases (with GitVersion, tag `0.1.0-alpha.1` as `0.1.0-rc.1`, make a patch commit, `0.1.0-alpha.2` is next instead of `0.1.0-rc.1.1`).

So, that said, I have a couple of questions.

First, is there any way to make the default prerelease versions increment the "primary" digit until a prerelease tag is encountered? So, for example, a series of commits and tags might look like this:

 - Tag head as `1.0.0`
 - New commit, auto versioned as `1.0.1-alpha.0`
 - New commit, auto versioned as `1.0.1-alpha.1`
 - Tag `1.0.1-rc.0`
 - New commit, auto versioned as `1.0.1-rc.0.1`
 - New commit, auto versioned as `1.0.1-rc.0.2`
 - Tag `1.0.1-rc.1`
 - Tag `1.0.1`

Instead of this:

 - Tag head as `1.0.0`
 - New commit, auto versioned as `1.0.1-alpha.0.0`
 - New commit, auto versioned as `1.0.1-alpha.0.1`
 - Tag `1.0.1-rc.0`
 - New commit, auto versioned as `1.0.1-rc.0.1`
 - New commit, auto versioned as `1.0.1-rc.0.2`
 - Tag `1.0.1-rc.1`
 - Tag `1.0.1`

And second, is there any way to easily query the minver-cli to check if the current version is a prerelease? Perhaps something equivalent to `gitversion -showvariable PreReleaseTag`?

Thanks again!

**Comment by AshleighAdams** [comment]
As a follow up, for the prerelease query, I think this is always guaranteed to be valid:

```sh
if [[ "$(minver --tag-prefix=v --verbosity=e)" == *"-"* ]]; then
  options+=("--prerelease")
fi
```

But for the first issue, I encountered this on the README:

>This will result in a post-RTM version of {major}.{minor}.{patch+1}-preview.{height}, e.g. 1.0.1-preview.1.

Did the behavior change at some point? This is what I want (number-wise).

**Comment by adamralph** [comment, marked as answer]
@AshleighAdams in answer to your first question, the answer is similar to this one — https://github.com/adamralph/minver/discussions/488#discussioncomment-339197. The behaviour, as far as I can remember has never changed. This was the design from the start.

Regarding your second question, checking the version for the presence of `-` should work as long as their is no build metadata, which could also contain `-`. To be absolutely safe, I would first split the string on `+` and then check for the presence of `-` in the first substring.

**Reply by AshleighAdams** [reply]
Yeah, I assumed this would be out of scope, so I decided to take my own shot at this, using the same principles that's not as minimal as MinVer, but nowhere near as as heavy as GitVersion.

Thanks for your response.
